rewrite-maven/src/main/java/org/openrewrite/maven/tree/MavenResolutionResult.java

```diff
@@ -69,12 +70,21 @@ public class MavenResolutionResult implements Marker {
     @With
     List<String> activeProfiles;
 
+    @With
+    Map<String, String> userProperties;
+
     public List<String> getActiveProfiles() {
         // for backwards compatibility with ASTs that were serialized before activeProfiles was added
         //noinspection ConstantValue
         return activeProfiles == null ? emptyList() : activeProfiles;
     }
 
+    public Map<String, String> getUserProperties() {
```

Comment:
An alternative to this would be to add a package private @JsonCreator annotated constructor with a nullable field, which dies the defaulting.

Having a comment which we can use to later find and remove that constructor (or this method) would also be nice.

Reply:
Yes I'd went for this approach as there was already a no args constructor with that creator annotation.
https://github.com/openrewrite/rewrite/pull/5114/files/72e88c7cc8b23ac8706a9fd353440fa73d13d3a7#diff-2cf9cbe4851aaf4690268bae848441a39ce782334d3e1350d1a5976076cc388eR39

Figured best then to copy the approach we already had for active profiles.